test/gtest/conv_embed_db.cpp

```diff
@@ -0,0 +1,243 @@
+#include <tuple>
+
+#include <miopen/miopen.h>
+#include <gtest/gtest.h>
+#include "conv_2d.hpp"
+#include "get_handle.hpp"
+
+using TestCase = std::tuple<std::vector<std::string>, std::string>;
+
+enum class Precision
+{
+    Float,
+    Half,
+    Int8,
+    BFloat16
+};
+
+std::string GetFloatArg()
+{
+    static const auto tmp = std::getenv("MIOPEN_TEST_FLOAT_ARG");
```

Comment:
This makes the test dependent on the environment variable provided by CTest. I think this is acceptable for now, but this is a workaround. What behavior we need want if test is run independently of CTest?

/cc @JehandadKhan

Reply:
This was added, specifically to accomodate the current test design which relies on CMake variables to determine which precisions should and should not run. By default ( ie when no vars are set) all the precisions should run.

Reply:
This was added, specifically to accomodate the current test design which relies on CMake variables to determine which precisions should and should not run. By default ( ie when no vars are set) all the precisions should run.

Reply:
> @JehandadKhan @alexandraBara
> 
> > By default ( ie when no vars are set) all the precisions should run.
> 
> But this is not implemented yet. Shall we do it right here or let's postpone?

@atamazov , right now, the gtest case is indeed controlled by environment variables as below since we are in a transit state from CTest to gTest. It appears by default, "--float" will be enabled

https://github.com/ROCmSoftwarePlatform/MIOpen/blob/54c586f34535e50b2dd56bcd78233b896a5050b5/test/CMakeLists.txt#L203-L227
 
 My understand is as we moving completely away from CTests, we can enable all the tests simply by removing the env var controlled function, i.e. IsTestRunWith()

https://github.com/ROCmSoftwarePlatform/MIOpen/blob/20532ed32950b6612ac29159fe14f50fcc5ad797/test/gtest/conv_embed_db.cpp#L116C1-L116C69